src/lib/helpers/buildPaginationHelper.ts

```diff
@@ -0,0 +1,183 @@
+        const isEllipsis = numOfPages >= 5 && currentPageNumber >= 5;
+        pageItems.push(
+            createPageItem(2, currentPageNumber, isEllipsis, urlPrefix, searchQuery)
+        );
+    }
+
+    // Add element at middle left position if applicable
+    if (
+        numOfPages >= 5 &&
+        currentPageNumber >= 4 &&
+        numOfPages - currentPageNumber >= 1
+    ) {
+        pageItems.push(
+            createPageItem(
+                currentPageNumber - 1,
+                currentPageNumber,
+                false,
+                urlPrefix,
+                searchQuery
+            )
+        );
+    }
+
+    // Add element at middle position if applicable
+    if (
+        numOfPages >= 5 &&
+        currentPageNumber >= 3 &&
+        numOfPages - currentPageNumber >= 2
+    ) {
+        pageItems.push(
+            createPageItem(
+                currentPageNumber,
+                currentPageNumber,
+                false,
+                urlPrefix,
+                searchQuery
+            )
+        );
+    }
+
+    // Add element at middle right position if applicable
+    if (
+        numOfPages >= 5 &&
+        currentPageNumber >= 2 &&
+        numOfPages - currentPageNumber >= 3
+    ) {
+        pageItems.push(
+            createPageItem(
+                currentPageNumber + 1,
+                currentPageNumber,
+                false,
+                urlPrefix,
+                searchQuery
+            )
+        );
+    }
+
+    // Add second-to-last element if applicable - possible ellipsis
+    if (numOfPages >= 4) {
+        const isEllipsis = numOfPages >= 5 && numOfPages - currentPageNumber >= 4;
+        pageItems.push(
+            createPageItem(
+                numOfPages - 1,
+                currentPageNumber,
+                isEllipsis,
+                urlPrefix,
+                searchQuery
+            )
+        );
+    }
+
+    // Add last element if applicable
+    if (numOfPages > 1) {
+        pageItems.push(
+            createPageItem(
+                numOfPages,
+                currentPageNumber,
+                false,
+                urlPrefix,
+                searchQuery
+            )
+        );
+    }
+
+    pagination.items = pageItems;
+    return pagination;
+};
+
+const createPageItem = (
+    pageNumber: number,
+    currentPageNumber: number,
+    isEllipsis: boolean,
+    prefix: string,
+    searchQuery: string
+): PageItem => {
+    if (isEllipsis) {
+        return {
+            ellipsis: true
+        };
+    }
+    return {
+        current: currentPageNumber === pageNumber,
+        number: pageNumber,
+        href: `${prefix}?page=${pageNumber}${searchQuery}`
+    };
+};
+
+export const paginationElement = (
+    page: number,
+    arrayLength: number,
+    search: string,
+    itemsPerPage: number
+): PaginationData | undefined => {
+    // Create pagination element to navigate pages
+    const numOfPages = Math.ceil(arrayLength / itemsPerPage);
```

Comment:
dont think you have to do all this, as we provide all the info in the rest response.

Reply:
as above

Reply:
Code refactored